src/Symfony/Bundle/WebProfilerBundle/Resources/views/Profiler/base_js.html.twig

```diff
@@ -235,77 +235,94 @@
         }
 
         {% if excluded_ajax_paths is defined %}
-            if (window.XMLHttpRequest && XMLHttpRequest.prototype.addEventListener) {
-                if (window.fetch) {
-                    var oldFetch = window.fetch;
-                    window.fetch = function () {
-                        var promise = oldFetch.apply(null, arguments);
-                        var stackElement = {
-                            loading: true,
-                            error: false,
-                            url: arguments[0],
-                            method: arguments[1].method,
-                            start: new Date()
-                        };
-                        requestStack.push(stackElement);
-                        promise.then(function (r) {
-                            stackElement.duration = new Date() - stackElement.start;
-                            stackElement.loading = false;
-                            stackElement.error = r.status < 200 || r.status >= 400;
-                            stackElement.statusCode = r.status;
-                            stackElement.profile = r.headers.get('x-debug-token');
-                            stackElement.profilerUrl = r.headers.get('x-debug-token-link');
-                            Sfjs.renderAjaxRequests();
-                        });
-                        Sfjs.renderAjaxRequests();
-                        return promise;
-                    };
-                }
-                var proxied = XMLHttpRequest.prototype.open;
 
-                XMLHttpRequest.prototype.open = function(method, url, async, user, pass) {
-                    var self = this;
+        if (window.fetch) {
+            var oldFetch = window.fetch;
+            window.fetch = function () {
+                if (!arguments[0].match(new RegExp({{ excluded_ajax_paths|json_encode|raw }}))) {
```

Comment:
applying the original fetch (and returning the promise) must still be made even when the URL is excluded from the profiling, otherwise you are breaking ``window.fetch``